conda_build/environ.py

```diff
@@ -191,6 +191,9 @@ def get_dict(m=None, prefix=None):
 
     elif sys.platform.startswith('linux'):      # -------- Linux
         d['LD_RUN_PATH'] = prefix + '/lib'
+        if cc.bits == 32:
+            d['CFLAGS'] = ' '.join([os.getenv('CFLAGS', ''), d.get('CFLAGS', ''), '-m32'])
+            d['CXXFLAGS'] = ' '.join([os.getenv('CXXFLAGS', ''), d.get('CXXFLAGS', ''), '-m32'])
 
```

Comment:
Also, if you're going to fix the linux-32 case, why not go ahead and fix it in the darwin case, just above.